### PR TITLE
#115 Fix/remove moment

### PR DESCRIPTION
**Pull request by adamdavies1**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Please also include any acceptance criteria if you have any.

- Removed remaining uses of moment

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce
the testing if necessary.

Before PR's can be merged they will need to be tested by QA and approved where
applicable. To flag the change to QA assign **@XGovFormBuilder/qa** as one of the reviewers.

- [ ] Runner and model had some changes, make sure the date fields are still working and no invalid dates can be put in
- [ ] Create DateTime component
- [ ] Create DateTime part component
- [ ] Check the DateTimePart component doesn't accept and invalid date e.g. 31st November

# Checklist:

- [x] My changes do not introduce any new linting errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and versioning
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto main and there are no code conflicts
- [ ] I have checked deployments are working in all environments
- [ ] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

**Review by endamccormack — APPROVED** [review]
looks good to me, worth @jenbutongit having a look before merge

**Review by jenbutongit — CHANGES_REQUESTED** [review]

File: runner/src/server/plugins/engine/components/DateTimePartsField.ts [hunk 1]
```diff
@@ -105,26 +104,38 @@ export class DateTimePartsField extends FormComponent {
...
-          payload[`${name}__year`],
-          payload[`${name}__month`] - 1,
-          payload[`${name}__day`],
-          payload[`${name}__hour`],
-          payload[`${name}__minute`],
-        ]).toDate()
-      : null;
+    const date = this.constructDateString(
+      payload[`${name}__year`],
+      payload[`${name}__month`],
```
[the remark this answers is not in the document]
Comment (adamdavies1) [reply runner/src/server/plugins/engine/components/DateTimePartsField.ts:113]: for this screenshot i put in the date 17/06/1992 11:11. Says its june so it appears to be working
image

**Comment by adamdavies1** [comment]
Putting this back in draft as I think i have found a bug

**Review by jenbutongit — APPROVED** [review]
thanks! 💪

**Pull request merged**